core/src/components/item/item.ios.vars.scss

```diff
@@ -25,7 +25,7 @@ $item-ios-paragraph-margin-start:            $item-ios-paragraph-margin-end !def
 $item-ios-paragraph-font-size:               dynamic-font(14px) !default;
 
 /// @prop - Color of the item paragraph
-$item-ios-paragraph-text-color:              rgba($text-color-rgb, .4) !default;
+$item-ios-paragraph-text-color:              var(--ion-text-color-step-550, #a3a3a3) !default;
```

Comment:
Not needed to ship this feature, so let me know if you'd prefer I add this in a separate PR.

Devs can add a `p` inside of an item. The problem is that `p` won't participate in high contrast because it's using the `--ion-text-color-rgb` var with an alpha channel instead of a step color. This change allows it to participate in high contrast. The values I chose are very close to the original computed values.